cluster_pack/filesystem.py

```diff
@@ -203,6 +230,19 @@ def put(self, filename: str, path: str, chunk: int = 2**16) -> None:
                     target.write(out)
         _preserve_acls(self.base_fs, filename, path)
 
+    def put_atomic(self,
+         source_path: str,
+         destination_path: str) -> None:
+        upload_dir = f"{destination_path}.{uuid.uuid4()}.tmp"
+        try:
+            self.put(source_path, upload_dir)

```

Comment:
[OPT] This operation will not be the one failing since we are writing on a random temporary location, only the mv op can have a concurrency issue.

Reply:
indeed, will move the try: